py_gtfs_rt_ingestion/py_gtfs_rt_ingestion/s3_utils.py

```diff
@@ -15,8 +18,14 @@ def file_list_from_s3(bucket_name: str,
     """
     logging.info("Getting files with prefix %s from %s" % (file_prefix,
                                                            bucket_name))
-    s3 = boto3.resource('s3')
-    bucket = s3.Bucket(bucket_name)
-
-    for bucket_object in bucket.objects.filter(Prefix=file_prefix):
-        yield (bucket_object.key, bucket_object.size)
+    s3_client = get_s3_client()
+    paginator = s3_client.get_paginator('list_objects_v2')
+    pages = paginator.paginate(
+        Bucket=bucket_name,
+        Prefix=file_prefix,
+    )
+    for page in pages:
+        if page['KeyCount'] == 0:
+            continue
```

Comment:
is this necessary? if the key count is 0, isn't `page['Contents']` just an empty list?

Reply:
In my testing, `KeyCount` is always in the response object. If `KeyCount == 0` then a `Contents` object is not sent.